NewsApp/NewsApp/Model/Services/NewsService.swift

```diff
@@ -10,17 +10,64 @@ import Foundation
 
 protocol NewsService {
     var newsArticles: [Article] { get }
-    func getTopHeadlines(for category: Category?, completion: @escaping () -> Void)
+    var parametersRequest: ParametersRequest { get }
+    func getArticles(completion: @escaping () -> Void)
 }
 
 class NewsServiceImpl: NewsService {
     var newsArticles: [Article] = []
+    let parametersRequest: ParametersRequest
     let apiClient = APIClient()
+
+    ///Initializers
+    init(parametersRequest: ParametersRequest) {
+        self.parametersRequest = parametersRequest
+    }
+    init() {
+        self.parametersRequest = ParametersRequest(requestType: .topHeadlines)
+    }
+    ///General
+    var requestType: RequestType {
+        return parametersRequest.requestType
+    }
+    ///Top-Headlines
     var country: Country? {
-        return PreferencesServiceImpl().getCountry()
+        return parametersRequest.country
+    }
+    var category: Category? {
+        return parametersRequest.category
+    }
+    ///Everything
+    var language: Language? {
+        return parametersRequest.language
+    }
+    var dateFrom: Date? {
+        return parametersRequest.from
+    }
+    var dateTo: Date? {
+        return parametersRequest.to
+    }
+    var query: String? {
+        return parametersRequest.query
+    }
+    var sortBy: SortBy? {
+        return parametersRequest.sortBy
+    }
+
+    func getArticles(completion: @escaping () -> Void) {
```

Comment:
d'après le nom de la fonction getArticles, on s'attend à ce que ça nous retourne des articles, du coup est-ce que les mettre dans la completion pourrait être intéressant ?

Reply:
fix in "Fix PR"